osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerParticipantsList.cs

```diff
@@ -43,6 +43,16 @@ public void TestAddUser()
             AddAssert("two unique panels", () => this.ChildrenOfType<ParticipantPanel>().Select(p => p.User).Distinct().Count() == 2);
         }
 
+        [Test]
+        public void TestAddNullUser()
+        {
+            AddAssert("one unique panel", () => this.ChildrenOfType<ParticipantPanel>().Select(p => p.User).Distinct().Count() == 1);
+
+            AddStep("add non-resolvable user", () => Client.AddNullUser(-3));
+
+            AddUntilStep("two unique panels", () => this.ChildrenOfType<ParticipantPanel>().Select(p => p.User).Distinct().Count() == 2);
+        }
```

Comment:
I think it would be better coverage if the null user is added in `TestSceneMultiplayerMatchSubScreen` instead, to ensure there shouldn't be any failures on any components in the whole screen.

Reply:
Can you give an example of what you are hoping to catch? I can't see any such usages.

Reply:
I'm saying that so that this test could track if a new component accesses `User` without null checking, one `Debug.Assert(User.User != null)` line would return this issue again.

Reply:
I'm not sure that would help. Components should be tested directly, not at a screen level.

It's enforced now via https://github.com/ppy/osu/pull/11341/files#diff-0db59d73b332bc157b321082be2954f26fe9275f47385a233cd5c5710f69808aR27 so can't happen at compile time going forward.